[Bug] Recent changes bring package downgrades in consuming libraries (not released yet)

**Which version of Microsoft.IdentityModel are you using?**
Note that to get help, you need to run the latest version. 
<!-- E.g. Microsoft.IdentityModel 6.14 -->

**Where is the issue?**
https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/3058 attempts to fix package downgrades but bring other problems. See

**Repro**

Attempt to build the Unified Builds with the dev version of IdentityModel.

**Expected behavior**
The build passes

**Actual behavior**
D:\a\_work\1\s\microsoft-identity-web\src\Microsoft.Identity.Web.Certificateless\Microsoft.Identity.Web.Certificateless.csproj : error NU1605: Warning As Error: Detected package downgrade: Microsoft.Extensions.Logging.Abstractions from 3.1.0 to 2.1.0. Reference the package directly from the project to select a different version.  [D:\a\_work\1\s\microsoft-identity-web\tests\E2E Tests\NET 7 tests\IntegrationTests\IntegrationTests.csproj]
D:\a\_work\1\s\microsoft-identity-web\src\Microsoft.Identity.Web.Certificateless\Microsoft.Identity.Web.Certificateless.csproj : error NU1605:  Microsoft.Identity.Web.Certificateless -> Microsoft.IdentityModel.JsonWebTokens 8.3.1 -> Microsoft.IdentityModel.Tokens 8.3.1 -> Microsoft.Extensions.Logging.Abstractions (>= 3.1.0)  [D:\a\_work\1\s\microsoft-identity-web\tests\E2E Tests\NET 7 tests\IntegrationTests\IntegrationTests.csproj]
D:\a\_work\1\s\microsoft-identity-web\src\Microsoft.Identity.Web.Certificateless\Microsoft.Identity.Web.Certificateless.csproj : error NU1605:  Microsoft.Identity.Web.Certificateless -> Microsoft.Extensions.Logging.Abstractions (>= 2.1.0) [D:\a\_work\1\s\microsoft-identity-web\tests\E2E Tests\NET 7 tests\IntegrationTests\IntegrationTests.csproj]
 

**Possible solution**
Differentiate the references based on the target framework (Ilike IdWeb and MISE do)

--- a/build/dependencies.props
+++ b/build/dependencies.props
@@ -5,13 +5,37 @@
     <BannedApiAnalyzersVersion>3.3.4</BannedApiAnalyzersVersion>
     <MicrosoftBclTimeProviderVersion>8.0.1</MicrosoftBclTimeProviderVersion>
     <MicrosoftCSharpVersion>4.5.0</MicrosoftCSharpVersion>
-    <MicrosoftExtensionsLoggingAbstractionsVersion>3.1.0</MicrosoftExtensionsLoggingAbstractionsVersion>
     <MicrosoftSourceLinkGitHubVersion>1.0.0</MicrosoftSourceLinkGitHubVersion>
     <NetStandardVersion>2.0.3</NetStandardVersion>
     <NewtonsoftVersion>13.0.3</NewtonsoftVersion>
     <SystemMemoryVersion>4.5.5</SystemMemoryVersion>
     <SystemSecurityCryptographyCngVersion>4.5.0</SystemSecurityCryptographyCngVersion>
     <SystemTextJson>8.0.5</SystemTextJson>
+  </PropertyGroup>
+
+
+<PropertyGroup Condition="'$(TargetFramework)' == 'net9.0'">
+    <MicrosoftExtensionsLoggingAbstractionsVersion>9.0.0</MicrosoftExtensionsLoggingAbstractionsVersion>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(TargetFramework)' == 'net8.0'">
+    <MicrosoftExtensionsLoggingAbstractionsVersion>8.0.0</MicrosoftExtensionsLoggingAbstractionsVersion>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(TargetFramework)' == 'net7.0'">
+    <MicrosoftExtensionsLoggingAbstractionsVersion>7.0.0</MicrosoftExtensionsLoggingAbstractionsVersion>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(TargetFramework)' == 'net6.0'">
+    <MicrosoftExtensionsLoggingAbstractionsVersion>6.0.0</MicrosoftExtensionsLoggingAbstractionsVersion>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(TargetFramework)' == 'net462'">   
+    <MicrosoftExtensionsLoggingAbstractionsVersion>6.0.0</MicrosoftExtensionsLoggingAbstractionsVersion>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(TargetFramework)' == 'netstandard2.0' Or '$(TargetFramework)' == 'net472'">
+    <MicrosoftExtensionsLoggingAbstractionsVersion>2.1.0</MicrosoftExtensionsLoggingAbstractionsVersion>
   </PropertyGroup>
 
 </Project>

